examples/annotated-http-service-kotlin/src/main/kotlin/example/armeria/server/annotated/kotlin/ContextAwareService.kt

```diff
@@ -0,0 +1,56 @@
+package example.armeria.server.annotated.kotlin
+
+import com.linecorp.armeria.server.ServiceRequestContext
+import com.linecorp.armeria.server.annotation.Get
+import com.linecorp.armeria.server.annotation.Param
+import com.linecorp.armeria.server.annotation.ProducesJson
+import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.withContext
+import org.slf4j.LoggerFactory
+import java.util.concurrent.Executors
+
+class ContextAwareService {
+
+    @Get("/foo")
+    @ProducesJson
+    suspend fun foo(@Param("name") name: String, @Param("id") id: Int): FooResponse {
+        log.info("Hello $name")
+        // Make sure that current thread is request context aware
+        ServiceRequestContext.current()
+
+        // Propagate armeria request context
+        withContext(ArmeriaRequestContext()) {
```

Comment:
Does a user need to do this? Can't Armeria make sure the function is run with a request context dispatcher?

Reply:
In this example, `BusinessLogic.blockingTask` is designed to executed in `myDispatcher` to limit the number of parallel executions.
As we cannot dispatch to a request context dispatcher, then I pass a request context as `ThreadContextElement`.

Possibly this is a rare case..